compiler warning on building src/test/test_bitcoin-validation_chainstate_tests.o

**Is your feature request related to a problem? Please describe.**

Warning on building using g++ 8.3.

```
 1 In file included from ./txdb.h:11,
 2                  from ./validation.h:26,
 3                  from ./test/util/chainstate.h:13,
 4                  from test/validation_chainstate_tests.cpp:11:
 5 ./chain.h: In member function ‘void validation_chainstate_tests::chainstate_update_tip::test_method()’:
 6 ./chain.h:422:27: warning: ‘background_cs’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 7          return vChain.size() > 0 ? vChain[vChain.size() - 1] : nullptr;
 8                 ~~~~~~~~~~~^~
 9 test/validation_chainstate_tests.cpp:109:18: note: ‘background_cs’ was declared here
10      CChainState* background_cs;
11                   ^~~~~~~~~~~~~
12
```

**Describe the solution you'd like**

Able to build without warnings.

**Describe alternatives you've considered**

N/A.

**Additional context**

`cd src/ && make test/test_bitcoin-validation_chainstate_tests.o`

Bisecting this warning, I ended up in revision 673a5bd3377929a0
by @jamesob.

Thank you very much and please excuse the noise.

--- a/src/test/validation_chainstate_tests.cpp
+++ b/src/test/validation_chainstate_tests.cpp
@@ -107,20 +107,21 @@ BOOST_FIXTURE_TEST_CASE(chainstate_update_tip, TestChain100Setup)
 
     curr_tip = ::g_best_block;
 
-    CChainState* background_cs;
-
     BOOST_CHECK_EQUAL(chainman.GetAll().size(), 2);
-    for (CChainState* cs : chainman.GetAll()) {
-        if (cs != &chainman.ActiveChainstate()) {
-            background_cs = cs;
+
+    CChainState& background_cs{*[&] {
+        for (CChainState* cs : chainman.GetAll()) {
+            if (cs != &chainman.ActiveChainstate()) {
+                return cs;
+            }
         }
-    }
-    BOOST_CHECK(background_cs);
+        assert(false);
+    }()};
 
     // Create a block to append to the validation chain.
     std::vector<CMutableTransaction> noTxns;
     CScript scriptPubKey = CScript() << ToByteVector(coinbaseKey.GetPubKey()) << OP_CHECKSIG;
-    CBlock validation_block = this->CreateBlock(noTxns, scriptPubKey, *background_cs);
+    CBlock validation_block = this->CreateBlock(noTxns, scriptPubKey, background_cs);
     auto pblock = std::make_shared<const CBlock>(validation_block);
     BlockValidationState state;
     CBlockIndex* pindex = nullptr;
@@ -133,15 +134,15 @@ BOOST_FIXTURE_TEST_CASE(chainstate_update_tip, TestChain100Setup)
         LOCK(::cs_main);
         bool checked = CheckBlock(*pblock, state, chainparams.GetConsensus());
         BOOST_CHECK(checked);
-        bool accepted = background_cs->AcceptBlock(
+        bool accepted = background_cs.AcceptBlock(
             pblock, state, &pindex, true, nullptr, &newblock);
         BOOST_CHECK(accepted);
     }
     // UpdateTip is called here
-    bool block_added = background_cs->ActivateBestChain(state, pblock);
+    bool block_added = background_cs.ActivateBestChain(state, pblock);
 
     // Ensure tip is as expected
-    BOOST_CHECK_EQUAL(background_cs->m_chain.Tip()->GetBlockHash(), validation_block.GetHash());
+    BOOST_CHECK_EQUAL(background_cs.m_chain.Tip()->GetBlockHash(), validation_block.GetHash());
 
     // g_best_block should be unchanged after adding a block to the background
     // validation chain.

